File: admin/src/main/java/ru/cg/cda/admin/service/RoleService.java
package ru.cg.cda.admin.service;

import java.util.List;

import ru.cg.cda.admin.dto.RoleDTO;


/**
 * @author Badamshin
 */
public interface RoleService {
  RoleDTO getRole(Long roleId, boolean withGroupIds);

  List<RoleDTO> getRoles();

  RoleDTO saveRole(RoleDTO roleDTO);

  void saveRoleGroups(Long roleId, List<Long> groupIds);
}

File: admin/src/main/frontend/app/scripts/app/service/RoleService.ts
'use strict';

module cda.app.service {

  import Role = cda.app.model.Role;

  export interface IRoleService {
    getRoles(): ng.IPromise<Role[]>;
    getRole(roleId: number, withGroupIds ?: boolean): ng.IPromise<Role>;
    saveRole(role: Role): ng.IPromise<Role>;
  }

  class RoleService implements IRoleService {
    static $inject = ['Restangular'];

    constructor(private restangular: restangular.IService) {
    }

    getRoles(): ng.IPromise<Role[]> {
      return this.restangular.all('role').getList();
    }

    getRole(roleId: number, withGroupIds: boolean = false): ng.IPromise<Role> {
      return this.restangular.one('role', roleId).get({"withGroupIds": withGroupIds});
    }

    saveRole(role: Role): ng.IPromise<Role> {
      return this.restangular.all('role').post(role);
    }
  }

  angular.module("adminApp").service("RoleService", RoleService);
}
